File: engine/model/src/main/scala/tools/packet/character/UpdateMount.scala
package tools.packet.character

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class UpdateMount(private var _characterId: Int, private var _mountLevel: Int, private var _mountExp: Int,
                  private var _mountTiredness: Int, private var _levelUp: Boolean) extends PacketInput {
  def characterId: Int = _characterId

  def mountLevel: Int = _mountLevel

  def mountExp: Int = _mountExp

  def mountTiredness: Int = _mountTiredness

  def levelUp: Boolean = _levelUp

  override def opcode(): SendOpcode = SendOpcode.SET_TAMING_MOB_INFO
}

File: engine/engine-model/src/main/java/tools/packet/character/UpdateMount.java
package tools.packet.character;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record UpdateMount(Integer characterId, Integer mountLevel, Integer mountExp, Integer mountTiredness,
                          Boolean levelUp) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SET_TAMING_MOB_INFO;
   }
}